### ISSUE TITLE
[BUG] rpcDoRequest: rpc error code 400: DC_ID_INVALID

### ISSUE THREAD

**Issue by SoapDev2018** [iyear/tdl#106]
**Describe the bug**
Downloading a file from a protected chat, via a DC1 account results in this issue.

**To Reproduce**
Join any protected DC5/DC2 chat with DC1 account. Try to download any file with `tdl dl -u "link" -n foobar`

**Expected behavior**
File starts downloading.

**Screenshots**
![image](https://user-images.githubusercontent.com/36785676/216443168-988844be-f1a1-4661-8684-ffa6a2e09aed.png)

**Version:**
```
Version: 0.7.0
Commit: f364ee5
Date: 2023-01-31T12:27:10Z

go1.18.10 windows/amd64
```

**Additional context**
Debug Logs:
https://paste.sh/UCjOU7Gx#rLuARax_Ur2Z2ZJTl_C1RXw2

**Comment by iyear** [comment]
Yes, this is the same issue as #93. But the upstream library does not expose the CDN download interface to the public, so this will require me to go to the upstream library and make some contributions before fixing the tdl.

**Comment by arisuryaputra** [comment]
Click Allow in Warning from Telegram Channel Chat

**Comment by PkParadox88** [comment]
For the 1st time telegram asked data export request and after allowing the tdl worked fine and started to download. But the next day when I work with tdl neither it asks warning in telegram nor the file could be downloaded. It gives me the above error only. No idea what is the actual issue and how to fix it.
